sponsorship_compassion/models/contracts.py

```diff
@@ -574,7 +559,7 @@ def write(self, vals):
             self.on_change_partner_correspondent_id()
             self.auto_correspondent_id()
 
-        return True
+        return super().write(vals)
```

Comment:
Now you changed the order of execution by moving the super call at the end. It means that the changes in the sponsorship will only be processed after all our code which in my opinion is wrong for the other methods that were called after the super like `_on_language_changed` and `update_number_sponsorships` at line 542. Those two should still be after the super.